src/curvefit/initializer/initializer.py

```diff
@@ -0,0 +1,352 @@
+        pass
+
+
+class JointPriorInitializerComponent(PriorInitializerComponent):
+    """
+    {begin_markdown JointPriorInitializerComponent}
+
+    # `curvefit.initializer.initializer.JointPriorInitializerComponent`
+    ## A joint prior initializer component for an initializer for a model
+
+    See [`PriorInitializerComponent`](PriorInitializerComponent.md).
+
+    ## Attributes
+
+    - `self.component_type (str)`: "joint", which is the type of component
+
+    ## Methods
+
+    See methods for [`PriorInitializerComponent`](PriorInitializerComponent.md).
+
+    {end_markdown PriorInitializerComponent}
+    """
+    def __init__(self):
+        super().__init__()
+        self.initializer_type = 'joint'
+
+
+class IndividualPriorInitializerComponent(PriorInitializerComponent):
+    """
+    {begin_markdown IndividualPriorInitializerComponent}
+
+    # `curvefit.initializer.initializer.IndividualPriorInitializerComponent`
+    ## An individual prior initializer component for an initializer for a model
+
+    See [`PriorInitializerComponent`](PriorInitializerComponent.md).
+
+    ## Attributes
+
+    - `self.component_type (str)`: "individual", which is the type of component
+
+    ## Methods
+
+    See methods for [`PriorInitializerComponent`](PriorInitializerComponent.md).
+
+    {end_markdown IndividualPriorInitializerComponent}
+    """
+    def __init__(self):
+        super().__init__()
+        self.initializer_type = 'individual'
+
+
+class LnAlphaBetaPrior(IndividualPriorInitializerComponent):
+    """
+    {begin_markdown LnAlphaBetaPrior}
+
+    # `curvefit.initializer.initializer.LnAlphaBetaPrior`
+    ## Gets information about a ln alpha-beta prior for model
+
+    See [`PriorInitializerComponent`](PriorInitializerComponent.md) for a description of a prior initializer.
+    This prior initializer component uses group-specific individual
+    model fits with alpha and beta parameters to get information
+    about what the mean and standard deviation for a functional prior on log(alpha * beta) should be. It uses
+    the empirical mean and standard deviation of log(alpha * beta), typically fit on only a subset of groups
+    that have more data and information about what alpha and beta should be.
+
+    **NOTE: Requires 'alpha' and 'beta' parameters in a ParameterSet along with a functional prior
+    called 'ln-alpha-beta'.**
+
+    ## Attributes
+
+    See attributes for [`IndividualPriorInitializerComponent`](IndividualPriorInitializerComponent.md).
+
+    ## Methods
+
+    See methods for [`IndividualPriorInitializerComponent`](IndividualPriorInitializerComponent.md).
+
+    {end_markdown LnAlphaBetaPrior}
+    """
+    def __init__(self):
+        super().__init__()
+
+    def _extract_prior(self, solver):
+        assert isinstance(solver, List)
+
+        alphas = np.array([])
+        betas = np.array([])
+
+        for sol in solver:
+
+            alpha_idx = sol.model.param_set.get_param_index('alpha')
```

Comment:
Should we access sol.model through get_model_instance() instead? I thought this wrapper was meant to check that the model exists.

Reply:
yes I think we should. @mbannick right now model is only stored in the base solver. I changed to that to avoid repeatedly pass model instance at all levels, therefore we have to access model through `get_model_instance()`. let me know if this makes sense!

Reply:
and like Aleksei mentioned it will check whether the model exists.

Reply:
Yes @jizezhang @aksholokhov we should definitely do that. I can fix this tomorrow.

Reply:
@jizezhang @aksholokhov fixed in commit 875961d